src/Essentials/src/FileSystem/FileSystem.ios.cs

```diff
@@ -158,68 +160,134 @@ protected override void Dispose(bool disposing)
 		}
 	}
 
-	class UIDocumentFileResult : FileResult
+	class UIDocumentFileResult : MediaFileResult
 	{
-		internal UIDocumentFileResult(NSUrl url)
-			: base()
+		UIDocument document;
+		NSUrl assetUrl;
+
+		internal UIDocumentFileResult(NSUrl assetUrl, string fileName)
 		{
-			var doc = new UIDocument(url);
-			FullPath = doc.FileUrl?.Path;
-			FileName = doc.LocalizedName ?? Path.GetFileName(FullPath);
+			this.assetUrl = assetUrl;
+			document = new UIDocument(assetUrl);
+			Extension = document.FileUrl.PathExtension!;
+			ContentType = GetMIMEType(document.FileType);
+			NameWithoutExtension = !string.IsNullOrWhiteSpace(fileName)
+				? Path.GetFileNameWithoutExtension(fileName)
+				: null;
+			Type = GetFileType(ContentType);
+			FileName = GetFileName(NameWithoutExtension, Extension);
 		}
 
 		internal override Task<Stream> PlatformOpenReadAsync()
-		{
-			Stream fileStream = File.OpenRead(FullPath);
+			=> Task.FromResult<Stream>(File.OpenRead(document.FileUrl.Path!));
 
-			return Task.FromResult(fileStream);
+		protected internal override void PlatformDispose()
+		{
+			document?.Dispose();
+			document = null;
+			assetUrl?.Dispose();
+			assetUrl = null;
+			base.PlatformDispose();
```

Comment:
id document and assetUrl are set by all the constructors, they do not need to be nullable and we won't need to add ? or set them to null. Looks like there is something funny with the nullability annotations in the code.